apidef/api_definitions.go

```diff
@@ -231,6 +231,70 @@ type ExtendedPathsSet struct {
 	Internal                []InternalMeta        `bson:"internal" json:"internal,omitempty"`
 }
 
+// GraphQLConfig is the root config object for GraphQL Middlewares
+type GraphQLConfig struct {
+	// BasePath is the base path for the GraphQL API, Subscriptions Endpoint as well as the GraphQLPlayground
+	BasePath   string            `bson:"base_path" json:"base_path"`
+	// GraphQLApi is the API specific configuration
+	GraphQLApi        GraphQLApi        `bson:"graphql_api" json:"graphql_api"`
+	// GraphQLPlayground is the Playground specific configuration
+	GraphQLPlayground GraphQLPlayground `bson:"playground" json:"playground"`
+}
+
+// GraphQLApi is the configuration for the GraphQL Middleware
+type GraphQLApi struct {
+	// Endpoint is, combined with the base path, the route which the GraphQL Middleware reacts to
+	Endpoint      string               `bson:"endpoint" json:"endpoint"`
+	// Schema is the GraphQL Schema exposed by the GraphQL API/Upstream/Engine
+	Schema        string               `bson:"schema" json:"schema"`
+	// Subscriptions is the configuration regarding websocket subscriptions
+	Subscriptions GraphQLSubscriptions `bson:"subscriptions" json:"subscriptions"`
+	// Execution defines the mode and configuration in which the GraphQL middleware should operate
+	Execution     GraphQLExecution `bson:"execution" json:"execution"`
+}
+
+// GraphQLSubscriptions is the configuration for websocket based GraphQL Subscriptions
+type GraphQLSubscriptions struct {
+	// Enabled defines if Subscriptions should be accepted
+	Enabled                               bool   `bson:"enabled" json:"enabled"`
+	// Endpoint is, combined with the base path, the route on which websocket upgrade requests will be accepted
+	Endpoint                              string `bson:"endpoint" json:"endpoint"`
+	// SecureWebsocketProtocolHeaderVariable is the SecWebsocketProtocolHeader variable expected by the websocket client
+	SecureWebsocketProtocolHeaderVariable string `bson:"secure_websocket_protocol_header_variable" json:"secure_websocket_protocol_header_variable"`
+}
+
+// GraphQLExecution defines the GraphQL Execution Mode as well as its configuration
+type GraphQLExecution struct {
+	// Mode is the mode in which the Middleware should operate
+	Mode       GraphQLExecutionMode       `bson:"mode" json:"mode"`
+	// Validation defines the behaviour regarding GraphQL request validation
+	Validation GraphQLExecutionValidation `bson:"validation" json:"validation"`
+	// Config is the GraphQLExecutionMode specific configuration object
+	Config     json.RawMessage            `bson:"config" json:"config"`
```

Comment:
It probably makes sense to have it as interface{} instead (like with do with metadata for all objects), otherwise, you will have to parse it each time, since it is just byte slice.